job-server-api/src/spark.jobserver/NamedObjectSupport.scala

```diff
@@ -0,0 +1,134 @@
+package spark.jobserver
+
+import akka.util.Timeout
+import java.lang.ThreadLocal
+import java.util.concurrent.atomic.AtomicReference
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.storage.StorageLevel
+import scala.concurrent.duration.Duration
+
+sealed trait NamedObject
+case class NamedRDD[T](rdd: RDD[T], storageLevel: StorageLevel) extends NamedObject
+case class NamedDataFrame(df: DataFrame) extends NamedObject
+
+abstract class NamedObjectPersister[O <: NamedObject] {
+  //TODO what is the context needed for? c: ContextLike,
+  def saveToContext(namedObj: O, name: String): Unit
+  //TODO - what is this for?
+  //def getFromContext(c: ContextLike, name: String): O
+  /**
+   * update reference to named object so that it does not get GCed
+   */
+  def refresh(namedObject: O): O
+  //TODO - what about 'unpersist' ?
```

Comment:
Hmm, the NamedObjectPersister is currently not informed when an object is destroyed. I moved the unpersist into the destroy method of the _NamedRdds implementation, but I think it might be better if we added a destroy (or removeFromContext) method to NameObjectPersister?